### PR TITLE
GUI for web streams/feeds management

### PR DESCRIPTION
**Pull request by SubJunk**
https://github.com/UniversalMediaServer/UniversalMediaServer/issues/1602

### CODE REVIEW

**Comment by valib** [comment]
@SubJunk good idea.

**Comment by SubJunk** [comment]
@valib I'm fairly happy with this GUI now, when you have time can you please test/review? No rush

File: src/main/java/net/pms/newgui/SpringUtilities.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+import javax.swing.SpringLayout;
+import java.awt.*;
+
+/**
+ * A 1.4 file that provides utility methods for
+ * creating form- or grid-style layouts with SpringLayout.
+ * These utilities are used by several programs, such as
+ * SpringBox and SpringCompactGrid.
+ */
+public class SpringUtilities {
```
Comment (valib) [inline src/main/java/net/pms/newgui/SpringUtilities.java]: @SubJunk why is this class here?

File: src/main/java/net/pms/newgui/SpringUtilities.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */ 
+
+package layout;
```
Comment (valib) [inline src/main/java/net/pms/newgui/SpringUtilities.java]: @SubJunk wrong package

File: src/main/java/net/pms/newgui/SharedContentTab.java [hunk 3]
```diff
@@ -0,0 +1,748 @@
...
+		@Override
+		public void setValueAt(Object aValue, int row, int column) {
+			Vector rowVector = (Vector) dataVector.elementAt(row);
+			rowVector.setElementAt(aValue, column);
+			fireTableCellUpdated(row, column);
+			updateWebContentModel();
+		}
+	}
+
+	public class TableMouseListener extends MouseAdapter {
```
Comment (valib) [inline src/main/java/net/pms/newgui/SharedContentTab.java]: @SubJunk you implemeted this listener and will you continue to add the feature when the mouse double click on the line will open the content panel to be able to edit the line setting? Now when user added a new line with wrong setting the only possibility is delete this wrong line and make the new one without possibility to edit them.

File: src/main/java/net/pms/newgui/SpringUtilities.java (hunk 1 above)
Comment (OP-Klaus) [reply src/main/java/net/pms/newgui/SpringUtilities.java]: Ah yeah they were added during an experiment with the GUI, thanks for removing

File: src/main/java/net/pms/newgui/SharedContentTab.java [hunk 4]
```diff
@@ -0,0 +1,748 @@
...
+
+		/* An attempt to set the correct row height adjusted for font scaling.
+		 * It sets all rows based on the font size of cell (0, 0). The + 4 is
+		 * to allow 2 pixels above and below the text. */
+		DefaultTableCellRenderer cellRenderer = (DefaultTableCellRenderer) sharedFoldersList.getCellRenderer(0,0);
+		FontMetrics metrics = cellRenderer.getFontMetrics(cellRenderer.getFont());
+		sharedFoldersList.setRowHeight(metrics.getLeading() + metrics.getMaxAscent() + metrics.getMaxDescent() + 4);
+		sharedFoldersList.setIntercellSpacing(new Dimension(8, 2));
+
+		JImageButton but = new JImageButton("button-add-folder.png");
```
Comment (valib) [inline src/main/java/net/pms/newgui/SharedContentTab.java]: @SubJunk should be chaged to something like "plus" image than the "folder" image.

File: src/main/java/net/pms/newgui/SpringUtilities.java (hunk 1 above)
Comment (SubJunk) [reply src/main/java/net/pms/newgui/SpringUtilities.java]: Ah yeah they were added during an experiment with the GUI, thanks for removing

File: src/main/java/net/pms/newgui/SharedContentTab.java (hunk 4 above)
Comment (SubJunk) [reply src/main/java/net/pms/newgui/SharedContentTab.java]: @valib good idea

File: src/main/java/net/pms/newgui/SharedContentTab.java (hunk 3 above)
Comment (SubJunk) [reply src/main/java/net/pms/newgui/SharedContentTab.java]: @valib yeah that sounds like a good idea, will do that

**Comment by SubJunk** [comment]
@valib editing is working now

**Comment by SubJunk** [comment]
Button is changed now too. But I just realised the tooltip text is wrong on the bottom panel and I ran out of time tonight

**Comment by valib** [comment]
@SubJunk I chaged the layout of the web content adding and editing panel. If you don't like it feel free to reset it to your original solution.

**Comment by SubJunk** [comment]
I agree it's better how you did it, let's leave that

**Comment by SubJunk** [comment]
@valib not sure if you have seen this https://github.com/UniversalMediaServer/UniversalMediaServer/issues/1612 but if you agree, I can change this PR to base on an `8.0.0` branch so we can merge the related branches into there instead of `master`

**Pull request merged**